Test/DependencyCollector/FunctionalTests/FuncTest/SqlTests.cs

```diff
@@ -336,6 +342,110 @@ public void TestSqlCommandExecuteXmlReaderFailed()
+            // instrumentation works only in case of profiler
+            if (RegistryCheck.IsStatusMonitorInstalled)
+            {
+                this.TestSqlConnectionExecute("SqlConnectionOpenAsyncAwait", true, false);
+            }
+        }
+
+        [DeploymentItem("..\\TestApps\\ASPX451\\App\\", DeploymentAndValidationTools.Aspx451AppFolder)]
+        [TestMethod]
+        public void TestSqlConnectionOpenAsyncAwaitException()
+        {
+            // instrumentation works only in case of profiler
+            if (RegistryCheck.IsStatusMonitorInstalled)
+            {
+                this.TestSqlConnectionExecute("SqlConnectionOpenAsyncAwait", false, false);
+            }
+        }
+
+        #endregion
+
+        private void TestSqlConnectionExecute(string type, bool success, bool async)
+        {
+            DeploymentAndValidationTools.Aspx451TestWebApplication.DoTest(
+                 application =>
+                 {
+                     string responseForQueryValidation = application.ExecuteAnonymousRequest("?type=" + type + "&success=" + success);
+
+                     //// The above request would have trigged RDD module to monitor and create RDD telemetry
+                     //// Listen in the fake endpoint and see if the RDDTelemtry is captured                      
+
+                     var allItems = DeploymentAndValidationTools.SdkEventListener.ReceiveAllItemsDuringTimeOfType<TelemetryItem<RemoteDependencyData>>(
+                         async ? DeploymentAndValidationTools.SleepTimeForSdkToSendAsyncEvents : DeploymentAndValidationTools.SleepTimeForSdkToSendEvents);
+                     var sqlItems = allItems.Where(i => i.data.baseData.type == "SQL").ToArray();
+                     if (success)
+                     {
+                         // no items should be sent in case of success connection to database
+                         Assert.AreEqual(0, sqlItems.Length, "Total Count of Remote Dependency items for SQL collected is wrong.");
+                     }
+                     else
+                     { 
+                         Assert.AreEqual(1, sqlItems.Length, "Total Count of Remote Dependency items for SQL collected is wrong.");
+
+                         this.Validate(
+                             sqlItems[0],
+                             ResourceNameSQLToWrongDatabase,
+                             "Open",
+                             TimeSpan.FromSeconds(20),
+                             successFlagExpected: success,
+                             sqlErrorCodeExpected: "-1",
+                             sqlErrorMessageExpected: "A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)");
```

Comment:
This seems to be great and accurate condition but somewhat error-prone, the test failures might be induced by the updates in .NET FW rather than the updates in our code. I remember we were validating such messages in APM agent and tests were sometimes broken after update 4.0->4.5 and 4.5->4.6. Maybe, (contains) to look for some token that will always logically be present in this message?